src/solver/linear/preconditioners/l1_gauss_seidel.jl

```diff
@@ -0,0 +1,310 @@
+####################################
+## L1 Gauss Seidel Preconditioner ##
+####################################
+
+## Structs & Constructors ##
+struct BlockPartitioning{Ti,Backend}
+    partsize::Ti # number of diagonals per partition
+    nparts::Ti # number of partitions
+    backend::Backend 
+end
+
+@doc raw"""
+    L1GSPreconditioner{Partitioning, VectorType}
+
+The ℓ₁ Gauss–Seidel preconditioner is a robust and parallel-friendly preconditioner, particularly effective for sparse and ill-conditioned systems.
+
+# Algorithm
+
+The L1-GS preconditioner is constructed by dividing the matrix into blocks `nparts`:
+- Let Ωₖ denote the block with index `k`.
+- For each Ωₖ, we define the following sets:
+    - Ωⁱ $:= \{j ∈ Ωₖ : i ∈ Ωₖ\}$ → the set of columns in the diagonal block for row i
+    - Ωⁱₒ $:= \{j ∉ Ωₖ : i ∈ Ωₖ\}$ →  the remaining “off-diagonal” columns in row i
+
+The preconditioner matrix $M_{ℓ_1}$  is defined as:
+```math
+M_{ℓ_1} = \mathcal{M}_H + D^{ℓ_1} = \text{diag}\{B_k + D_k^{ℓ_1}\} \\
+B_k = A_{kk}
+```
+Where $D^{ℓ_1}$ is a diagonal matrix with entries
+
+```math
+d_{ii}^{ℓ_1} = \sum_{j ∈ Ωⁱₒ} |a_{ij}|
+```
+
+# Fields
+- `partitioning`: Encapsulates partitioning data (e.g. nparts, partsize, backend).
+- `B`: Vector of diagonal entries (used in normalization).
+- `D`: Vector of ℓ₁ row norms of off-diagonal entries.
+
+
+# Reference
+Baker, A. H., Falgout, R. D., Kolev, T. V., & Yang, U. M. (2011).  
+*Multigrid Smoothers for Ultraparallel Computing*,  
+SIAM J. Sci. Comput., 33(5), 2864–2887.  
+[https://doi.org/10.1137/100798806](https://www.osti.gov/servlets/purl/1117969)
```

Comment:
See https://juliadocs.org/DocumenterCitations.jl/stable/syntax/ where the references file is given here https://github.com/Abdelrahman912/Thunderbolt.jl/blob/2532179e3b6fcc87241868ce17cb00f10282cf25/docs/src/assets/references.bib

I.e.
```suggestion
[Baker, A. H., Falgout, R. D., Kolev, T. V., & Yang, U. M. (2011).  
*Multigrid Smoothers for Ultraparallel Computing*,  
SIAM J. Sci. Comput., 33(5), 2864–2887.](@cite BakFalKolYan:2011:MSU)
```

and add the entry for `BakFalKolYan:2011:MSU` into the bibtex file correctly.